<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Offers a range of methods for simulating and imputing missing values 
             in different types of omics data such as genomics, transcriptomics, 
             proteomics, and metabolomics. Provides tools for comparing and 
             evaluating the performance of imputation methods in the terms of 
             accuracy and computation time.">
<title>The Most Robust Wrapper for Missing Values Imputation in ‘Omics’ Data • imputomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="The Most Robust Wrapper for Missing Values Imputation in ‘Omics’ Data">
<meta property="og:description" content="Offers a range of methods for simulating and imputing missing values 
             in different types of omics data such as genomics, transcriptomics, 
             proteomics, and metabolomics. Provides tools for comparing and 
             evaluating the performance of imputation methods in the terms of 
             accuracy and computation time.">
<meta property="og:image" content="http://biogenies.info/imputomics/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">imputomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/group_analysis.html">A Quick Guide for the imputomics shiny app</a>
    <a class="dropdown-item" href="articles/quick_start.html">A Quick Guide on imputomics</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/biogenies/imputomics/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.svg" class="logo" alt=""><h1 id="imputomics-">imputomics <a class="anchor" aria-label="anchor" href="#imputomics-"></a>
</h1>
</div>
<p><em>imputomics</em> is an R package and a shiny web server designed to simulate and impute missing values. It offers 42 algorithms for imputing missing values, especially in different types of ‘-omics’ data such as genomics, transcriptomics, proteomics, and metabolomics. imputomics provides a user-friendly interface that allows users to simulate missing values based on different distributions and impute missing values using state-of-the-art methods.</p>
<div class="section level2">
<h2 id="key-features">Key Features:<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Imputation methods: imputomics offers the biggest collection of imputation methods for different types of omics data, including k-nearest neighbors (KNN), random forests, expectation-maximization (EM) algorithm, and principal components analysis (PCA) and many others.</p></li>
<li><p>Performance evaluation: imputomics facilitates evaluating the performance of imputation methods. Users can evaluate imputation accuracy and compare different methods using metrics such as root mean squared error (RMSE), mean absolute error (MAE), and coefficient of determination (R-squared).</p></li>
<li><p>Simulation of missing values: imputomics provides a variety of options for simulating missing values, including missing completely at random (MCAR), missing at random (MAR), and missing not at random (MNAR) mechanisms. Users can specify the percentage of missing values and the distribution from which the missing values are generated.</p></li>
</ol>
</div>
</div>
<div class="section level1">
<h1 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h1>
<p>This repository contains the data and code necessary to reproduce the results from the paper <em>imputomics: comprehensive missing data imputation for metabolomics data</em>. It uses <a href="https://CRAN.R-project.org/package=renv" class="external-link">renv</a> package to assure the reproducibility. As <em>imputomics</em> implements lots of missing value imputations methods from other R packages.</p>
<div class="section level2">
<h2 id="web server">Web server<a class="anchor" aria-label="anchor" href="#web server"></a>
</h2>
<p>The <em>imputomics</em> can be accessed through our <a href="https://biogenies.info/imputomics-ws/" class="external-link">web server</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><em>imputomics</em> is available on <a href="https://github.com/BioGenies/imputomics" class="external-link">GitHub</a></p>
<p>To install <em>imputomics</em> you need to have <em>R</em> version &gt;= 4.2.0.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BioGenies/imputomics"</span><span class="op">)</span></span></code></pre></div>
<p>Sometimes, not all packages can be installed on the first try. In this case, consider re-running the <em>install_github</em> function.</p>
<div class="section level3">
<h3 id="docker">Docker<a class="anchor" aria-label="anchor" href="#docker"></a>
</h3>
<p>To enhance the reproducibility of <em>imputomics</em>, we share it also as a <a href="https://github.com/rocker-org" class="external-link">rocker-based</a> container. The docker manifest is available in the <em>imputomics</em> repository: <a href="https://github.com/BioGenies/imputomics/blob/main/Dockerfile_imputomics" class="external-link uri">https://github.com/BioGenies/imputomics/blob/main/Dockerfile_imputomics</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>To reproduce our environment you need to git clone our repo and activate renv.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/BioGenies/imputomics.git</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">renv</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/renv/reference/activate.html" class="external-link">activate</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">renv</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/renv/reference/restore.html" class="external-link">restore</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h3>
<p>Q: I am receiving the following error message: “Error: HTTP error 403. API rate limit exceeded for [my IP]”. A: Due to its comprehensiveness, <em>imputomics</em> downloads many packages from GitHub, which may lead to exceeding the limit of GitHub API queries. Please consider setting the GitHub API token with usethis::create_github_token().</p>
</div>
</div>
<div class="section level2">
<h2 id="run-imputomics">Run imputomics<a class="anchor" aria-label="anchor" href="#run-imputomics"></a>
</h2>
<p>To run <em>imputomics</em> type the following command into an R console.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">imputomics</span><span class="fu">::</span><span class="fu"><a href="reference/imputomics_gui.html">imputomics_gui</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<!-- # How to cite -->
<div class="section level4">
<h4 id="how-to-cite">How to cite?<a class="anchor" aria-label="anchor" href="#how-to-cite"></a>
</h4>
<p>Jarosław Chilimoniuk, Krystyna Grzesiak, Jakub Kała, Dominik Nowakowski, Adam Krętowski, Rafał Kolenda, Michał Ciborowski, Michał Burdukiewicz (2023). imputomics: web server and R package for missing values imputation in metabolomics data, <em>Bioinformatics</em>, <a href="https://doi.org/10.1093/bioinformatics/btae098" class="external-link">10.1093/bioinformatics/btae098</a>.</p>
</div>
<div class="section level4">
<h4 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h4>
<p>If you have any questions, suggestions or comments, contact <a href="mailto:michalburdukiewicz@gmail.com">Michal Burdukiewicz</a>.</p>
</div>
<div class="section level4">
<h4 id="funding-and-acknowledgements">Funding and acknowledgements<a class="anchor" aria-label="anchor" href="#funding-and-acknowledgements"></a>
</h4>
<p>We want to thank the Clinical Research Centre (Medical University of Białystok) members for fruitful discussions. K.G. wants to acknowledge grant no. 2021/43/O/ST6/02805 (National Science Centre). M.C. acknowledges grant no. B.SUB.23.533 (Medical University of Białystok). The study was supported by the Ministry of Education and Science funds within the project ‘Excellence Initiative - Research University’. We also acknowledge the Center for Artificial Intelligence at the Medical University of Białystok (funded by the Ministry of Health of the Republic of Poland).</p>
<p><img src="reference/figures/funding.png" style="height: 90px"></p>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/biogenies/imputomics/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/biogenies/imputomics/issues" class="external-link">Report a bug</a></li>
<li><a href="https://biogenies.info/imputomics-ws/" class="external-link">imputomics web server</a></li>
<li><a href="http://170.187.185.89:3838/imputomics/" class="external-link">imputomics web server backup</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing imputomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Michał Burdukiewicz <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8926-582X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Krystyna Grzesiak <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0003-2581-7722" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jarosław Chilimoniuk <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0001-5467-018X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jakub Kała <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-7187-6988" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Dominik Nowakowski <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-3515-4320" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Michał Burdukiewicz, Krystyna Grzesiak, Jarosław Chilimoniuk, Jakub Kała, Dominik Nowakowski.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
